pkg/controlplane/control/manager.go

```diff
@@ -61,87 +135,167 @@ func (m *Manager) AddImport(ctx context.Context, imp *v1alpha1.Import) error {
 					TargetPort: intstr.FromInt32(int32(imp.Spec.TargetPort)),
 				},
 			},
-			Type:     v1.ServiceTypeClusterIP,
 			Selector: map[string]string{"app": dpapp.Name},
+			Type:     v1.ServiceTypeClusterIP,
 		},
 	}
 
-	var oldService v1.Service
-	var create bool
-	err := m.client.Get(
-		ctx,
-		types.NamespacedName{
+	if err := m.addService(ctx, systemService); err != nil {
+		return err
+	}
+
+	if imp.Namespace == m.namespace {
+		return nil
+	}
+
+	userService := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
 			Name:      imp.Name,
 			Namespace: imp.Namespace,
 		},
-		&oldService)
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			return err
-		}
+		Spec: v1.ServiceSpec{
+			ExternalName: fmt.Sprintf("%s.%s.svc.cluster.local", serviceName, m.namespace),
+			Type:         v1.ServiceTypeExternalName,
+		},
+	}
+
+	return m.addService(ctx, userService)
+}
+
+// DeleteImport removes the listening socket of a previously imported service.
+func (m *Manager) DeleteImport(ctx context.Context, name types.NamespacedName) error {
+	m.logger.Infof("Deleting import '%s/%s'.", name.Namespace, name.Name)
+
+	// delete user service
+	errs := make([]error, 2)
+	errs[0] = m.deleteService(ctx, name)
```

Comment:
consider initializing the `errs` array with a single error (from the `deleteService` in L171) only if it is not a nil error?
Then `append`/`errors.Join()` another error to it in line 176 if that fails.

Reply:
This requires me to add `if` clauses. And `errors.Join` already handles `nil`